### ISSUE TITLE
Using lookup('template', ...) to update s3_bucket policy does not work

### ISSUE THREAD

**Issue by pako-pl** [ansible/ansible#14529]
##### Issue Type:
- Bug Report
##### Ansible Version:

ansible 2.0.1.0
##### Summary:

Trying to update S3 bucket policy using lookup('template', ...) does not work:

```
- s3_bucket:
    name: test-bucket
    state: present
    policy: "{{ lookup('template','policy.j2') }}"
  delegate_to: management-host.net
```

I get the following error:

```
FAILED! => {"failed": true, "msg": "an unexpected type error occurred. Error was sequence item 0: expected string, dict found"}
```

It looks like the resulting JSON code gets interpreted by Ansible as a dictionary. When I save the generated JSON output to a temporary file:

```
- template: src=policy.j2 dest=/tmp/policy.json

```

everything works fine, as I can use lookup('file', '/tmp/policy.json') instead of lookup('template','policy.j2'):

```
- s3_bucket:
    name: test-bucket
    state: present
    policy: "{{ lookup('file','/tmp/policy.json') }}"
  delegate_to: management-host.net
```

The contents of the policy.j2 template:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "AllowAll",
            "Effect": "Allow",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": "arn:aws:s3:::*"
        }
    ]
}
```

**Comment by bcoca** [comment]
this should prevent the transformation:

`policy: "{{ lookup('template','policy.j2')|to_json }}"`

**Comment by pako-pl** [comment]
Thanks for the suggestion. Unfortunately, to_json didn't make any difference - I still see the same error:

```
{"failed": true, "msg": "an unexpected type error occurred. Error was sequence item 0: expected string, dict found"}
```

**Comment by bcoca** [comment]
different error than what that fixes, closing this in favor of dupe #14541 which explains the cause
